### ISSUE TITLE
Improper subtitle vertical sizing/positioning 

### ISSUE THREAD

**Issue by squarebracket** [bitmovin/bitmovin-player-ui#61]
The commits merged in from #30 introduced some problems with captions overlapping each other vertically in some conditions. For example, scaling down the window vertically:

![screenshot from 2017-10-16 12-52-08](https://user-images.githubusercontent.com/127774/31624540-0fc8346e-b271-11e7-988e-9210cea798ba.png)

Additionally, applying a black background causes lots of problems even when text is spaced out enough, since the text boxes are so close to one another:
![screenshot from 2017-10-16 12-59-05](https://user-images.githubusercontent.com/127774/31624787-d3fcae8c-b271-11e7-8d68-9cb6128fc550.png)
![screenshot from 2017-10-16 12-58-42](https://user-images.githubusercontent.com/127774/31624792-d6e27dfc-b271-11e7-8b54-0acc0c9a115e.png)

The subtitles should really be spaced out enough to allow for styling, and the vertical size of the video should be taken into account when sizing the subtitles.

**Comment by protyposis** [comment]
CEA-608 subtitle positioning was generally designed for a 4:3 display so it's rather difficult to get it render correctly in arbitrary aspect ratios. If we decrease the height, then the width is too narrow.

Do you need the positioning, or would it be an option for you if you could turn off the positioning and have the subtitles rendered as in previous UI versions?

**Comment by squarebracket** [comment]
Positioning would obviously be preferable since there is important information contained in the positioning, such as who is speaking.

I'm not really sure why you think this has to do with the aspect ratio? It clearly happens for 4:3 feeds:
![screenshot from 2017-10-16 13-37-38](https://user-images.githubusercontent.com/127774/31626337-541f55b0-b277-11e7-97af-97aef249ebf9.png)

Unless you mean it's difficult to calculate the size for subtitles for arbitrary `<video>` sizes. Though since the sizing works properly horizontally, I feel like it wouldn't be too hard to extend it to work vertically too -- you just use the smaller dimension as the anchor for the calculation.

**Comment by protyposis** [comment]
Yes the current implementation is definitely not ideal... I'll look into improvements, we'll find a solution.

**Comment by protyposis** [comment]
So, I think the overlapping text issue is about to be solved.

### Old

![cea-old2](https://user-images.githubusercontent.com/189372/31681385-eec29474-b32b-11e7-88c0-9c4bfd4ba1b0.gif)
![cea-old2_bg](https://user-images.githubusercontent.com/189372/31681391-f46e3f72-b32b-11e7-9f66-9a4cb01d5754.gif)

### New
![cea-new2](https://user-images.githubusercontent.com/189372/31681413-023be154-b32c-11e7-84ce-dd7608f43bc9.gif)
![cea-new2_bg](https://user-images.githubusercontent.com/189372/31681419-077ab140-b32c-11e7-848c-a60d73ec9517.gif)

What do you think? That's about as good as it will get. As you can see you can add a background color behind the text, but it's still not possible to pad the background (that will still lead to overlapping backgrounds). Adjusting the text size to accomodate the padding screws up the CEA-608 character grid so you'll have to disable that for CEA-608 captions via CSS. You can still keep the padding for non-CEA subtitles.

**Comment by squarebracket** [comment]
Did a checkout to `8826a96`, looks like now the positioning is off:
![screenshot from 2017-10-17 17-41-43](https://user-images.githubusercontent.com/127774/31691002-7ace4194-b362-11e7-89b2-ce89fccd68fa.png)

**Comment by squarebracket** [comment]
roll-up is also causing problems: ![screenshot from 2017-10-17 17-51-28](https://user-images.githubusercontent.com/127774/31691352-d1b9faa6-b363-11e7-9376-34671a54b2ca.png)

**Comment by protyposis** [comment]
Works correctly for me. Please make sure you don't have any interfering styles, especially since this PR changes the default style so it won't work correctly with an old CSS.

![image](https://user-images.githubusercontent.com/189372/31692191-a66634fc-b34d-11e7-9b54-33ce63301cdc.png)

**Comment by protyposis** [comment]
Positions were `undefined` in some cases, causing positioning issues. It is yet to be determined it that is a player or stream issue, but I anyway added a fix into the UI to always display them correctly.

![image](https://user-images.githubusercontent.com/189372/31692740-f29f0b6c-b34f-11e7-9b8d-9b307381979f.png)

**Comment by squarebracket** [comment]
> Positions were `undefined` in some cases
It's probably the CEA608 parser.

> Please make sure you don't have any interfering styles
```css
.bmpui-ui-skin-modern .bmpui-ui-subtitle-overlay .bmpui-ui-subtitle-label {
    background-color: black;
    color: white;
}
```

I rebuilt using `a72cf76`. Still see the issues. I guess I'll have to take a look at the code.
